### PR TITLE
added: docs support for aliyun oss 

### LINKED ISSUE #285: add documentation for aliyun oss storage
Documentation for changes in [dvc#1961](https://github.com/iterative/dvc/pull/1961).

### LINKED ISSUE #339: auto file deletion on cloning in windows
I tried to clone this repo on multiple machines running on windows platform, but due to some notations in the format of <filename>:<additiontag>.md it was automatically deleted and instead a cache file was downloaded creating havoc.

### PR DESCRIPTION
**Pull request by sanidhyamangal**
This is a pr for issue #285. Document support in remote_add and remote_modify is provided along with src/features.js and  src/Diagram/index.js

Fixes #285
Fixes #339

### CODE REVIEW

**Comment by sanidhyamangal** [comment]
@shcheklein I have updated some entries for sidebar.json kindly review my latest changes.

**Comment by sanidhyamangal** [comment]
@shcheklein  based on our discussion on renaming file names I have submitted a new issue #339, kindly review it.

File: static/docs/commands-reference/remote_add.md [hunk 1]
```diff
@@ -249,6 +249,46 @@ container doesn't already exist, it will be created automatically.
...
+
+### Click for Aliyun OSS
+
+First you need to setup OSS storage on Aliyun Cloud and then use s3 style url for oss storage and make endpoint a configurable value, an example is shown below:
+```
+$ dvc remote add myremote oss://my-bucket/path
+```
+To set key id, key secret and endpoint you need to use modify command from dvc, a sample ussge is show below:
+```
+$ dvc remote modify myremote oss_key_id my-key-id
```
Comment (shcheklein) [inline static/docs/commands-reference/remote_add.md]: _important_: we need to use `--local`. It's a **very bad practice** to put keys into a config file that goes into Git. Is there something similar to `~/.aws/credentials` for OSS - similar to S3?

File: static/docs/commands-reference/remote_add.md [hunk 2]
```diff
@@ -249,6 +249,46 @@ container doesn't already exist, it will be created automatically.
...
+```
+$ export OSS_ACCESS_KEY_ID="my-key-id"
+$ export OSS_ACCESS_KEY_SECRET="my-key-secret"
+$ export OSS_ENDPOINT="endpoint"
+```
+
+#### Test your oss storage using docker
+
+Start a container running an oss emulator.
+```
```
Comment (shcheklein) [inline static/docs/commands-reference/remote_add.md]: use `dvc` syntax highlighter "```dvc"

Comment (shcheklein) [reply static/docs/commands-reference/remote_add.md]: for all command line code blocks

File: static/docs/commands-reference/remote_add.md (hunk 1 above)
Comment (sanidhyamangal) [reply static/docs/commands-reference/remote_add.md]: @nanaya-tachibana Can you please throw some light on this?

**Comment by sanidhyamangal** [comment]
@shcheklein I have completed requested changes kindly review them.

File: static/docs/commands-reference/remote_add.md [hunk 3]
```diff
@@ -252,6 +252,46 @@ variables:
...
+You can also set enviornment variables and use them later, to set enviornment variables use following enviormnent variables:
+```dvc
+$ export OSS_ACCESS_KEY_ID="my-key-id"
+$ export OSS_ACCESS_KEY_SECRET="my-key-secret"
+$ export OSS_ENDPOINT="endpoint"
+```
+
+#### Test your oss storage using docker
+
+Start a container running an oss emulator.
```
Comment (shcheklein) [inline static/docs/commands-reference/remote_add.md]: add an extra new line before and after code blocks for readability

Comment (sanidhyamangal) [reply static/docs/commands-reference/remote_add.md]: @shcheklein  done.

**Comment by sanidhyamangal** [comment]
@shcheklein  I have added the ``--local`` flag in the code blocks and used prettier for wrapping the codes.

File: static/docs/commands-reference/remote_add.md (hunk 1 above)
Comment (sanidhyamangal) [reply static/docs/commands-reference/remote_add.md]: @shcheklein I have added ``--local`` flag in all the code blocks

File: static/docs/commands-reference/remote_add.md [hunk 4]
```diff
@@ -252,6 +252,57 @@ variables:
...
+$ dvc remote add myremote oss://my-bucket/path
+```
+
+To set key id, key secret and endpoint you need to use modify command from dvc, 
+a sample ussge is show below:
+
+```dvc
+$ dvc remote modify myremote oss_key_id my-key-id --local
+$ dvc remote modify myremote oss_key_secret my-key-secret --local
+$ dvc remote modify myremote oss_endpoint endpoint --local
```
Comment (shcheklein) [inline static/docs/commands-reference/remote_add.md]: 1. let's use `dvc remote modify myremote --local ...` it's easier to read
2. no need for `--local` for the endpoint one

File: static/docs/commands-reference/remote_add.md [hunk 5]
```diff
@@ -252,6 +252,57 @@ variables:
...
+
+First you need to setup OSS storage on Aliyun Cloud and then 
+use s3 style url for oss storage and make endpoint a configurable value, 
+an example is shown below:
+
+```dvc
+$ dvc remote add myremote oss://my-bucket/path
+```
+
+To set key id, key secret and endpoint you need to use modify command from dvc,
```
Comment (shcheklein) [inline static/docs/commands-reference/remote_add.md]: dvc -> DVC

File: static/docs/commands-reference/remote_modify.md [hunk 6]
```diff
@@ -228,6 +228,30 @@ For more information about the variables DVC supports, please visit
...
 </details>
 
+<details>
+
+### Click for Aliyun OSS available options
+
+- `oss_key_id` - oss key id to use to access a remote.
+
+  ```dvc
+    $ dvc remote modify myremote oss_key_id my-key-id --local
```
Comment (shcheklein) [inline static/docs/commands-reference/remote_modify.md]: remove indents in code blocks

File: static/docs/commands-reference/remote_add.md (hunk 5 above)
Comment (sanidhyamangal) [reply static/docs/commands-reference/remote_add.md]: @shcheklein  Done.

File: static/docs/commands-reference/remote_modify.md (hunk 6 above)
Comment (sanidhyamangal) [reply static/docs/commands-reference/remote_modify.md]: @shcheklein Done. Indents were added to match previous sections in the file. Though I have removed indents in mine.

File: static/docs/commands-reference/remote_add.md (hunk 4 above)
Comment (sanidhyamangal) [reply static/docs/commands-reference/remote_add.md]: @shcheklein updated positions of flags in my latest commits.

**Comment by shcheklein** [comment]
Thanks, @sanidhyamangal ! 🎉

**Pull request merged**